File: Trees/invertBinaryTree.js
class {
  constructor(value) {
    this.value = value
    this.left = null
    this.right = null
  }
}

function invertBinaryTree(tree) {
  let queue = [tree]

  while (queue.length) {
    let current = queue.shift()

    if (current === null) continue;

    swapNodes(current)

    queue.push(current.left)
    queue.push(current.right)
  }
}

function swapNodes(tree) {
  let left = tree.left
  tree.left = tree.right
  tree.right = left
}

File: Trees/invert_binary_tree.py
# Given the root of a binary tree, invert the tree, and return its root.

# Input: root = [4,2,7,1,3,6,9]
# Output: [4,7,2,9,6,3,1]

# Input: root = [4,2,7,1,3,6,9]
# Output: [4,7,2,9,6,3,1]

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def invertTree(self, root: TreeNode) -> TreeNode:
    if not root:
        return None
    
    temp = root.left
    root.left = root.right
    root.right = temp

    self.invertTree(root.left)
    self.invertTree(root.right)

    return root
    